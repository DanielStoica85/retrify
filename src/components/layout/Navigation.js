import React, { Component } from 'react';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    Container
} from 'reactstrap';

import SignedInLinks from './SignedInLinks';
import SignedOutLinks from './SignedOutLinks';

import { connect } from 'react-redux';

export class Navigation extends Component {
    state = {
        isOpen: false
    };

    toggle = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    };

    render() {
        const { auth } = this.props;
        const links = auth.uid ? <SignedInLinks /> : <SignedOutLinks />;
        return (
            <div>
                <Navbar color="dark" dark expand="sm" className="mb-5">
                    <Container>
                        <NavbarBrand href="/">Retrify</NavbarBrand>
                        <NavbarToggler onClick={this.toggle} />
                        <Collapse isOpen={this.state.isOpen} navbar>
                            <Nav className="ml-auto" navbar>
                                {links}
                            </Nav>
                        </Collapse>
                    </Container>
                </Navbar>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        auth: state.firebase.auth
    };
};

export default connect(mapStateToProps)(Navigation);
