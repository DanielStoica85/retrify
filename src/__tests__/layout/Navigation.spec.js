import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '../../components/layout/Navigation';

describe('Navigation component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<Navigation />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
