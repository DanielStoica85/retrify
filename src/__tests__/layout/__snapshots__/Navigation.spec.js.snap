// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<div>
  <nav
    className="mb-5 navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Retrify
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/register"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/retro"
              onClick={[Function]}
            >
              New Retro
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              Log Out
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              <span
                className="badge badge-danger badge-pill"
              >
                D.S.
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
